apps/dashboard/src/components/workflow-editor/steps/chat/chat-editor-preview.tsx

```diff
@@ -0,0 +1,116 @@
+import { CSSProperties, useEffect, useRef, useState } from 'react';
+import { RiChat1Fill } from 'react-icons/ri';
+import { type StepDataDto, type WorkflowResponseDto } from '@novu/shared';
+
+import { Code2 } from '@/components/icons/code-2';
+import { Accordion, AccordionContent, AccordionItem, AccordionTrigger } from '@/components/primitives/accordion';
+import { Button } from '@/components/primitives/button';
+import { Editor } from '@/components/primitives/editor';
+import { loadLanguage } from '@uiw/codemirror-extensions-langs';
+import { useEditorPreview } from '@/components/workflow-editor/steps/use-editor-preview';
+import { ChatPreview } from '@/components/workflow-editor/steps/chat/chat-preview';
+import { TabsSection } from '@/components/workflow-editor/steps/tabs-section';
+import { InlineToast } from '@/components/primitives/inline-toast';
+
+const getInitialAccordionValue = (value: string) => {
+  try {
+    return Object.keys(JSON.parse(value)).length > 0 ? 'payload' : undefined;
+  } catch (e) {
+    return undefined;
+  }
+};
+
+type ChatEditorPreviewProps = {
+  workflow: WorkflowResponseDto;
+  step: StepDataDto;
+  formValues: Record<string, unknown>;
+};
+
+const extensions = [loadLanguage('json')?.extension ?? []];
+
+export const ChatEditorPreview = ({ workflow, step, formValues }: ChatEditorPreviewProps) => {
```

Comment:
It would be nice if we can refactor some of the previews and commonize it within one component. I tried that but it seems like more work and adjustments are needed for the in-app.